Add an item appraisal cartridge for PDAs

chrome-cirrus opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
Adds an item appraisal cartridge for PDAs. Supports left-click interaction to appraise items while the program is active, and keeps a rolling history of those results visible in the PDA interface. Right-click appraisal utility verb is available for cases like bags or deep fryers that the PDA would be placed into on left-click, but those results won't be stored in history due to implementation horrors (see comments in code). Cribs heavily from the Netprobe cartridge for UI. 
<!-- What did you change in this PR? -->
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

## Why / Balance
Right now with very few exceptions a medical PDA is the obvious best choice for loadouts, and most of the alternatives have little to no additional functionality. More choices that do something beyond just having a different sprite is fun.
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->

## How to test
Start up a local instance, spawn in an appraisal cartridge and whatever PDAs you think look coolest. Install the cartridge and exercise the program on the items in the sandbox zone, verify things function as expected. 
<!-- Describe the way it can be tested -->

## Media
![appapp1](https://github.com/user-attachments/assets/46bc6630-1298-4033-b19d-13cf5eb0b2e8)
![appapp2](https://github.com/user-attachments/assets/238826dc-5c3a-4cd2-9fab-05db1428ec7f)

<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.
-->

## Requirements
<!-- 
Due to influx of PR's we require to ensure that PR's are following the correct guidelines.

Please take a moment to read these if its your first time.

Check the boxes below to confirm that you have in fact seen these (put an X in the brackets, like [X]):
-->
- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
<!--
List any breaking changes, including namespace, public class/method/field changes, prototype renames; and provide instructions for fixing them. This will be pasted in #codebase-changes.
-->

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up. Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog.
-->
:cl:
- add: Added an appraisal program cartridge for PDAs

dvir001 commented:
Dont install the PDA app into any PDA
Were working on moving all PDA "programs" to a new loadout tab so people can get any pda with any program. (Health analyzer. music etc)

You can leave it not mapped into any PDA if you add the program and we can add it to new loadouts after.

chrome-cirrus commented:
Oh badass! I'll remove the PDA additions then.

dvir001 commented:
All new frontier code has to be under _NF also, not just yaml

chrome-cirrus commented:
Alright, removed the pre-installation in cargo/QM PDAs. Scooted the files around and updated namespace in a few places as needed. Did a component addition thing to make the utility verb on right-click work (still doesn't integrate with the PDA UI part, but that's a better experience than just not having it at all).

chrome-cirrus commented:
Tested this on 'regular' PDA variants, as well as on the medical one that still has hardcoded components for added function. It doesn't interfere with the medical one, but you will get both an appraisal and medical scan on someone when you left click them with a medical PDA while the appraisal program is active. Harmless enough, and will get fixed when the medical scanner stuff is itself moved into a cartridge.

dvir001 commented:
https://github.com/space-wizards/space-station-14/pull/32450

Step closer to moving into a loadout

dvir001 commented on Resources/Locale/en-US/_NF/cargo/price-gun-component.ftl:
```diff
@@ -16,3 +16,6 @@ pirate-contraband-price-gun-pricing-result = Skizzit's says {THE($object)} {$pri
 pirate-contraband-price-gun-pricing-result-none = Skizzit's doesn't have an entry for {THE($object)}.
+
+# Appraisal cartridge
+appraisal-program-name = Appraisal App Plus
```
Move to cartridges.ftl under _NF (Make a new file)

chrome-cirrus replied:
Ah, thanks. Missed that bit, will fix it.

chrome-cirrus replied:
Alright, moved that to an appropriate new file. Also made the PDA UI column labels localized instead of hard coded.

Cheackraze reviewed (approved):
for now, i would say you can put some inside of the astro-vend vending machine list. easy spot to put it where we can just remove it later if need be

chrome-cirrus commented:
> for now, i would say you can put some inside of the astro-vend vending machine list. easy spot to put it where we can just remove it later if need be

Sounds good! Just pushed a commit up that puts it on the list for the regular and POI Astro Vends.

chrome-cirrus commented:
> for now, i would say you can put some inside of the astro-vend vending machine list. easy spot to put it where we can just remove it later if need be

Sounds good! Just pushed up a commit that adds it to the regular and POI Astro Vend variants.

![image](https://github.com/user-attachments/assets/c734e53b-47b0-4ac1-a8e3-3c67f87fe8a2)

ErhardSteinhauer reviewed (approved):
Tested in game, works
